src/CommunityToolkit/Xamarin.CommunityToolkit/Views/DockLayout.shared.cs

```diff
@@ -0,0 +1,164 @@
+        /// but this can be disabled using the LastChildFill.
+        /// </summary>
+        public bool LastChildFill
+        {
+            get => (bool)GetValue(LastChildFillProperty);
+            set => SetValue(LastChildFillProperty, value);
+        }
+
+        protected override void LayoutChildren(double x, double y, double width, double height)
+        {
+            var i = 0;
+
+            foreach (var child in Children)
+            {
+                if (child.IsVisible)
+                {
+                    i++;
+
+                    var sizeRequest = child.Measure(width, height, MeasureFlags.IncludeMargins);
+
+                    double childX = 0;
+                    double childY = 0;
+                    var request = sizeRequest.Request;
+                    var childWidth = Math.Min(width, request.Width);
+                    var childHeight = Math.Min(height, request.Height);
+
+                    var lastItem = i == Children.Count;
+                    if (lastItem & LastChildFill)
+                    {
+                        LayoutChildIntoBoundingRegion(child, new Rectangle(x, y, width, height));
+                        return;
+                    }
+
+                    switch (GetDock(child))
+                    {
+                        case Dock.Left:
+                            {
+                                childX = x;
+                                childY = y;
+                                childHeight = height;
+                                x += childWidth;
+                                width -= childWidth;
+                                break;
+                            }
+                        case Dock.Top:
+                            {
+                                childX = x;
+                                childY = y;
+                                childWidth = width;
+                                y += childHeight;
+                                height -= childHeight;
+                                break;
+                            }
+                        case Dock.Right:
+                            {
+                                childX = x + width - childWidth;
+                                childY = y;
+                                childHeight = height;
+                                width -= childWidth;
+                                break;
+                            }
+                        case Dock.Bottom:
+                            {
+                                childX = x;
+                                childY = y + height - childHeight;
+                                childWidth = width;
+                                height -= childHeight;
+                                break;
+                            }
+                        default:
+                            {
+                                goto case Dock.Left;
```

Comment:
can we avoid **goto**? :) just move **default** to the appropriate case (and the same for another switch

Reply:
Sure!. Changed.